src/app/tests/TestSchemaUtils-Device.cpp

```diff
@@ -356,6 +358,71 @@ void TestSchemaUtils::TestSchemaUtilsEncAndDecList(nlTestSuite * apSuite, void *
     }
 }
 
+void TestSchemaUtils::TestSchemaUtilsEncAndDecIteratableList(nlTestSuite * apSuite, void * apContext)
+{
+    CHIP_ERROR err;
+    chip::app::TestSchemaUtils *_this = static_cast<chip::app::TestSchemaUtils *>(apContext);
+
+    _this->mpSuite = apSuite;
+    _this->SetupBuf();
+    
+    {    
+        chip::app::Cluster::TestCluster::StructC::Type sc;
+        uint8_t d[5];
+        chip::app::Cluster::TestCluster::StructA::Type e[5];
+
+
+        for (size_t i = 0; i < ArraySize(d); i++) {
+            d[i] = (uint8_t)i;
+        }
+
+        sc.a = 20;
+        sc.b = 30;
+        sc.c.x = 99;
+        sc.c.y = 17;
+        sc.d = chip::Span<uint8_t>{d};
+        sc.e = chip::Span<chip::app::Cluster::TestCluster::StructA::Type>{e};
+
+        for (size_t i = 0; i < ArraySize(e); i++) {
+            e[i].x = (uint8_t)i;
+            e[i].y = (uint8_t)i;
+        }
+
+        err = chip::app::EncodeSchemaElement(sc, _this->mWriter, TLV::AnonymousTag);
+        NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+    }
+
+    _this->DumpBuf();
+    _this->SetupReader();
+
+    {
+        chip::app::Cluster::TestCluster2::IteratableStructC::Type sc;
+        int i;
+
+        err = sc.Decode(_this->mReader);
+        NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+
+        NL_TEST_ASSERT(apSuite, sc.a == 20);
+        NL_TEST_ASSERT(apSuite, sc.b == 30);
+        NL_TEST_ASSERT(apSuite, sc.c.x == 99);
+        NL_TEST_ASSERT(apSuite, sc.c.y == 17);
+
+        i = 0;
+        for (auto iter = sc.d.begin(); !(iter == sc.d.end()); ++iter) {
```

Comment:
The net stack cost here is a little high FYI - on top of the stack usage of `sc` above, this for loop results in the instantiation of `IteratableList::Iterator` twice, which results in a cost of 64 bytes + 2 * sizeof list member, which in this case, is 1 byte.